expansion/src/main/java/no/unit/nva/expansion/model/ExpandedResource.java

```diff
@@ -74,12 +73,44 @@ public static ExpandedResource fromPublication(RawContentRetriever uriRetriever,
         return attempt(() -> objectMapper.treeToValue(sortedJson, ExpandedResource.class)).orElseThrow();
     }
 
-    private static JsonNode addFields(String json, Publication publication) {
-        var sortedJson = strToJsonWithSortedContributors(json);
-        injectHasFileEnum(publication, (ObjectNode) sortedJson);
-        expandLicenses(sortedJson);
-        return sortedJson;
-    }
+  private static JsonNode addFields(String json, Publication publication) {
+    var sortedJson = strToJsonWithSortedContributors(json);
+    injectHasFileEnum(publication, (ObjectNode) sortedJson);
+    expandLicenses(sortedJson);
+    injectAnthologyRelation(publication, (ObjectNode) sortedJson);
+    return sortedJson;
+  }
+
+  private static void injectAnthologyRelation(Publication publication, ObjectNode sortedJson) {
+    Optional.ofNullable(publication.getEntityDescription())
+        .flatMap(desc -> Optional.ofNullable(desc.getReference()))
+        .ifPresent(
+            reference -> {
+              var instanceType = reference.getPublicationInstance();
+              var publicationContext = reference.getPublicationContext();
+              var isPartOfAnthology =
+                  publicationContext instanceof Anthology
+                      && (instanceType instanceof ChapterArticle
+                          || instanceType instanceof ChapterConferenceAbstract
+                          || instanceType instanceof ChapterInReport);
+
+              if (instanceType instanceof BookAnthology) {
+                addJoinField(sortedJson, "hasParts", null);
```

Comment:
Why?

Reply:
It can improve readability and add more semantic meaning, i.e. `"joinField"` could be the value of constant`RELATION_NODE`. This is not important, chose what you prefer 😄

Reply:
Fair enough, found some more descriptive names now :)